pkg/ui/workspaces/cluster-ui/src/api/txnInsightsApi.ts

```diff
@@ -511,20 +501,21 @@ function formatTxnInsightsRow(row: TxnInsightsResponseRow): TxnInsightEvent {
 
 export type TxnInsightsRequest = {
   txnExecutionID?: string;
+  txnFingerprintID?: string;
   start?: moment.Moment;
   end?: moment.Moment;
 };
 
 export function getTxnInsightsApi(
   req?: TxnInsightsRequest,
 ): Promise<TxnInsightEvent[]> {
-  const request = makeInsightsSqlRequest([
-    createTxnInsightsQuery({
-      execID: req?.txnExecutionID,
-      start: req?.start,
-      end: req?.end,
-    }),
-  ]);
+  const filters: TxnQueryFilters = {
+    start: req?.start,
+    end: req?.end,
+    execID: req?.txnExecutionID ? req.txnExecutionID : null,
+    fingerprintID: req?.txnFingerprintID ? req.txnFingerprintID : null,
```

Comment:
nit: This can be simplified to `req?.txFingerprintID`. You don't need the ternary operator here, the `?` will make the value null